app/views/comments/_form.html.erb

```diff
@@ -0,0 +1,6 @@
+<%= form_for [commentable, Comment.new] do |f| %>
```

Comment:
if the user is not logged in, ask for email, first name, last name too. with the email, sign up the user and then make the comment for the user.

Reply:
what about password?

Reply:
we just generate a random password for them. we then send an email welcoming the user to the site and send link to set their password. Some guide on how to do this here: https://github.com/plataformatec/devise/wiki/How-To:-Email-only-sign-up